src/app/layouts/default-layout/default-layout.component.ts

```diff
@@ -19,22 +19,31 @@ import { of } from 'rxjs';
   templateUrl: './default-layout.component.html',
   styleUrl: './default-layout.component.css',
 })
-export class DefaultLayoutComponent {
+export class DefaultLayoutComponent implements OnInit {
   private authService = inject(AuthService);
   private router = inject(Router);
 
+  private authLoading$ = toObservable(this.authService.isLoading);
+
   isInitializing = signal(false);
 
-  constructor() {
-    if (this.router.url === '/') {
-      this.handleRootNavigation();
-    }
+  ngOnInit() {
+    this.router.events
```

Comment:
We have a lot of logic to automagically send people to different routes based on user/admin status now. I wonder if this is getting a bit overcomplicated, and we should instead think about a landing page that shows links to the user/admin dashboards. Not something we need to cover in this PR, just wanted to bring it up.